src/libsync/owncloudpropagator.cpp

```diff
@@ -599,6 +598,22 @@ bool PropagateDirectory::scheduleNextJob()
     if (_state == NotYetStarted) {
         _state = Running;
 
+        // at the begining of the Directory Job, add all Container jobs to the main queue map
+        QMutableVectorIterator<PropagatorJob *> containerJobsIterator(_containerJobs);
+        while (containerJobsIterator.hasNext()) {
+            PropagatorJob * job = containerJobsIterator.next();
+            // TODO: The call for getJobPredicateValue should recursively traverse all kids in _containerJobs of _containerjobs,
+            // in order to update their predicates. Root job should call updatePredicates in owncloudPropagator as a last step before run
+            // , and save flag _folderPredicateUpdated = true
+             _subJobs.insertMulti(job->getJobPredicateValue(), job);
```

Comment:
This one should basicaly be moved to the moment before root get started, and the function _root->updateContainersPredicates() should perform left-deep traversal of all the folders with changed files ( _containerJobs), and be added to _subJobs list with <job->getJobPredicateValue(), job>. This has to be done like that, because files are being added to containers and we have to update _subJobs list of all folders starting from deepest ones.